compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -616,7 +617,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         );
 
         let mut err = match *error {
-            SelectionError::Unimplemented => {
+            SelectionError::Unimplemented | SelectionError::NotTransmutable { .. } => {
```

Comment:
I would probably just handle this branch separately.

Reply:
Actually, see comment below. I'm not convinced this needs a separate `SelectionError`.

Reply:
I had it separate but then I realized I would have to add a bunch of the code from this one in order to create the full error (for example, including the code block and pointing to say "required by a bound..."...

Reply:
Hm, this function is getting somewhat out of control. I wonder if you could factor it out in some way to make it more manageable...

Anyways, if you go ahead with my advice below and just use `Unimplemented` then it's kind of moot, but really this could use some investment in clean-up if you're up for it.

Reply:
I kept a separate variant, and was able to show the rest of the original error message by only copying over the necessary logic from the `Unspecified` case. But I can put up another PR that refactors the `Unspecified` case so we can reuse pieces and have it not be all in one giant code block.